### DISCUSSION TITLE
Issue running workflow to obtain voltage data from Measurement computing Daq (MccDaq)

### DISCUSSION THREAD

**Discussion by cameycalzolano** [bonsai-rx/bonsai#1235] [Q&A] [answered]
Hi all, 

I am trying to set up a basic workflow to collect voltage data from a measurement computing DAQ. I am using the AnalogInput sink (included in the MccDaq library that I have installed) to record 3 different channels - I’ve included a snapshot of what the node looks like. I attempt to isolate the data output (assuming these are the voltage values from the daq input) by right clicking the AnalogInput node and selecting “Output (OpenCV.Net.Mat)” and from there selecting “data”. I then connect this to a MatrixWriter node, but get an error when I attempt to run the workflow,  reading “no method overload found for the given arguments.” Note that I have “OpenCV.Net.Mat” selected for the TypeMapping property in the data output. Is there something I am missing? Should I not be pulling out the “data” output of the AnalogInput and instead just connecting MatrixWriter to AnalogInput directly?  I know someone asked a similar question in [this thread](https://groups.google.com/g/bonsai-users/c/DZzGDy6y41Y/m/hZKyGe1aAAAJ), but it doesn't look like think it was ever answered.

Also, note that we are using Bonsai32 as it is the only version of Bonsai compatible with the driver (CL eye multicam driver) that we will be using for our cameras (PS3Eye Camera).

Any input is much appreciated - thank you so much!

![Workflow-Capture](https://user-images.githubusercontent.com/114766484/213555309-32fc0254-f8d1-46ce-8ef5-9ff8773c074d.PNG)

**Comment by glopesdev** [comment, marked as answer]
`MatrixWriter` can handle a `Mat` input directly, so it should work if you just remove the middle node.

The `Data` property is actually a pointer to the raw buffer memory location, and usually only needed for lower-level manipulations.
